### PR TITLE
Update plugins to consume remote_sources parameter

### PR DESCRIPTION
**Pull request by mkosiarc**
CLOUDBLD-4449

Instead of receiving individual parameters related to
remote_sources such as build_args, configs, icm_url, url
plugins will now receive only one parameter that is a list of dicts
with these values.

For now only first element of the list will be used -
remote_sources[0] as multiple remote_sources are not yet supported.

This is in preparation for completing cachito-multiple-upstreams epic
CLOUDBLD-2838

Signed-off-by: mkosiarc <mkosiarc@redhat.com>

# Maintainers will complete the following section

- [x] Commit messages are descriptive enough
- [x] Code coverage from testing does not decrease and new code is covered
- [n/a] JSON/YAML configuration changes are updated in the relevant schema
- [n/a] Changes to metadata also update the documentation for the metadata
- [n/a] Pull request has a link to an osbs-docs PR for user documentation updates
- [n/a] New feature can be disabled from a configuration file

### CODE REVIEW

File: atomic_reactor/plugins/pre_add_image_content_manifest.py
Comment (tkdchen) [inline atomic_reactor/plugins/pre_add_image_content_manifest.py:98]: Suggest to create a "remote source" object.

**Review by tkdchen — COMMENTED** [review]
typo in commit message `receieve`.

File: atomic_reactor/plugins/pre_add_image_content_manifest.py
Comment (mkosiarc) [reply atomic_reactor/plugins/pre_add_image_content_manifest.py:98]: That is not a bad idea, but this was the decision taken in the design

Comment (chmeliik) [reply atomic_reactor/plugins/pre_add_image_content_manifest.py:98]: Improvements over the design are always welcome :wink: 

I think the param has to be a dict though. Correct me if I'm wrong, but I think all plugin params must be JSON-compatible objects because that's how osbs-client passes them to atomic-reactor. You could convert the param into some kind of "remote source" object inside the plugin though, I'll leave that up to you.

File: atomic_reactor/plugins/pre_add_image_content_manifest.py
Comment (chmeliik) [inline atomic_reactor/plugins/pre_add_image_content_manifest.py:146]: I don't think you need to check for `remote_sources[0]["icm_url"] is None`. I think checking that `remote_sources` is not None should be good enough. You could even prevent icm_url from being None in the schema (atomic_reactor/schemas/user_params.json)

The same pattern appears in other plugins too, checking that `remote_sources` is not None should usually be good enough.

File: atomic_reactor/plugins/pre_add_image_content_manifest.py
Comment (MartinBasti) [reply atomic_reactor/plugins/pre_add_image_content_manifest.py:98]: Yup, it has to be JSONizable (if that is a word, JSON serializable)

Comment (mkosiarc) [reply atomic_reactor/plugins/pre_add_image_content_manifest.py:97]: I am going to keep it as dict for now, unless someone is strongly against.

File: atomic_reactor/plugins/pre_add_image_content_manifest.py
Comment (mkosiarc) [reply atomic_reactor/plugins/pre_add_image_content_manifest.py]: So if remote_sources was specified, we would expect that all values will be present and not None for each remote source?

From https://github.com/containerbuildsystem/atomic-reactor/blob/master/atomic_reactor/plugins/pre_download_remote_source.py#L107

It looks like configs parameter might be empty?

Comment (chmeliik) [reply atomic_reactor/plugins/pre_add_image_content_manifest.py]: Yeah, remote_sources will be a list of objects with
```python
build_args: dict[str, str]
configs: str  # this is actually a url
request_id: int
url: str
name: Optional[str]
```
i.e. `name` is the only one which can ever be `None`

File: atomic_reactor/utils/cachito.py
Comment (rcerven) [inline atomic_reactor/utils/cachito.py:230]: this should be list of request_ids

File: atomic_reactor/plugins/pre_add_image_content_manifest.py
Comment (rcerven) [inline atomic_reactor/plugins/pre_add_image_content_manifest.py:137]: this should already pass all request_ids 

since we are allowing only 1 request from pre_resolve_remote_source it won't be issue

File: atomic_reactor/plugins/pre_resolve_remote_source.py
Comment (rcerven) [inline atomic_reactor/plugins/pre_resolve_remote_source.py:120]: we don't need source_request anymore to get 'id'

as that id is also already in remote_sources

Comment (mkosiarc) [reply atomic_reactor/plugins/pre_resolve_remote_source.py]: updated

File: atomic_reactor/plugins/pre_add_image_content_manifest.py
Comment (mkosiarc) [reply atomic_reactor/plugins/pre_add_image_content_manifest.py]: updated

File: atomic_reactor/utils/cachito.py
Comment (mkosiarc) [reply atomic_reactor/utils/cachito.py]: updated

**Review by rcerven — APPROVED** [review]

**Review by lkolacek — APPROVED** [review]

File: test.sh
Comment (MartinBasti) [inline test.sh:30]: :eyes:

Comment (mkosiarc) [reply test.sh:30]: otherwise tests would fail, this will be removed before merging.

Comment (MartinBasti) [reply test.sh:30]: just making note :)

**Comment by mkosiarc** [comment]
rn updated : Update plugins to consume remote_sources parameter

**Pull request merged**
